### PR TITLE
ckpt_activation docs

### PR DESCRIPTION
**Pull request by Oldify**
Docs for ckpt activation in pytorch & fairscale

### CODE REVIEW

**Comment by Jack47** [comment]
awesome, will merge and use  [./memory-efficiency](https://github.com/jack47/hack-SysML/blob/master/memory-efficiency) dir. You may also refer to my notes: [gradient checkpointing](https://github.com/jack47/hack-SysML/blob/master/memory-efficiency/gradient-checkpointing.md#L32)
